### DISCUSSION TITLE
import file (images) tags/location from metadata

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#5995] [Feature Requests]
[here](https://docs.directus.io/concepts/files/#file-fields) it says:

> Tags — Pulled from the file metadata if available
Location — Pulled from the file metadata if available

but that doesn't seem to be the case. Is it something that has still to be implemented?

**Comment by rijkvanzanten** [comment]
@wc-matteo What image are you testing with?

**Reply by wc-matteo** [reply]
I've tested with multiple and also checked the code. Should be in [here](https://github.com/directus/directus/blob/e2c9e15a981bd7034b1c5dad839a9816148aa594/api/src/services/files.ts#L29), right?

**Reply by rijkvanzanten** [reply]
Yeah, just curious what sort of metadata your image files have, so i can ensure it's reading the correct tags 👍🏻

**Reply by wc-matteo** [reply]
### test image
https://www.dropbox.com/s/1dn7y2qhrwkybqz/test_metadata_image.jpg

### wrote metadata in lightroom (except location)

![Screenshot 2021-06-02 211206](https://user-images.githubusercontent.com/10562635/120539066-86b80280-c3e7-11eb-943c-2eed0fd1de51.png)

![Screenshot 2021-06-02 211229](https://user-images.githubusercontent.com/10562635/120539088-8b7cb680-c3e7-11eb-857f-08c28629c8f6.png)

![Screenshot 2021-06-02 211247](https://user-images.githubusercontent.com/10562635/120539097-8ddf1080-c3e7-11eb-87a9-83580360be1b.png)

### windows file explorer properties

![Screenshot 2021-06-02 211114](https://user-images.githubusercontent.com/10562635/120539160-a4856780-c3e7-11eb-8eb6-8ffb78af8540.png)

![Screenshot 2021-06-02 211139](https://user-images.githubusercontent.com/10562635/120539187-ab13df00-c3e7-11eb-8144-a0cb1192edc5.png)

### test metadata web tool
http://exif.regex.info/exif.cgi

![Screenshot 2021-06-02 211050](https://user-images.githubusercontent.com/10562635/120539327-c54dbd00-c3e7-11eb-8a1b-6324eef44af8.png)

### directus v9.0.0-rc.70

![image](https://user-images.githubusercontent.com/10562635/120539636-0e057600-c3e8-11eb-8c13-8271c91ad4bc.png)

**Reply by benhaynes** [reply]
We scrape multiple formats, perhaps the IPTC is overriding EXIF? We'll have to look into this. We might want to "upgrade" this to a GitHub issue (actual bug report) so it doesn't get lost amidst feature requests.

**Reply by rijkvanzanten** [reply]
Updated the service to pull the headline and keywords correctly. We'll revisit location later, as Exif only provides that as lat+long, where Directus currently allows any string under location.

https://github.com/directus/directus/pull/6035

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
